q2cli/tests/test_cli.py

```diff
@@ -282,6 +285,33 @@ def test_verbose_shows_stacktrace(self):
         self.assertEqual(result.exit_code, 1)
         self.assertIn('Traceback (most recent call last)', result.output)
 
+    def test_input_conversion(self):
+        obj = QIIME2Type(IntSequence1.to_ast(), repr(IntSequence1))
+
+        with self.assertRaisesRegex(click.exceptions.BadParameter,
+                                    f'{self.tempdir!r} is not a QIIME 2 '
+                                    'Artifact'):
+            obj._convert_input(self.tempdir, None, None)
+
+        with self.assertRaisesRegex(click.exceptions.BadParameter,
+                                    "'x' is not a valid filepath"):
+            obj._convert_input('x', None, None)
+
+        obj._convert_input = \
+            unittest.mock.Mock(side_effect=click.exceptions.
+                               BadParameter('There was not enough space left '
+                                            f'on {self.tempdir!r} to extract '
+                                            'the artifact '
+                                            f'{self.artifact1_path!r}. (Try '
+                                            'setting $TMPDIR to a directory '
+                                            'with more space, or increasing '
+                                            f'the size of {self.tempdir!r})'))
+        with self.assertRaisesRegex(click.exceptions.BadParameter,
+                                    f'{self.tempdir!r}.*'
+                                    f'{self.artifact1_path!r}.*'
+                                    f'{self.tempdir!r}'):
+            obj._convert_input(self.artifact1_path, None, None)
```

Comment:
This last assertion isn't doing anything as the SUT (system under test) has been entirely replaced by the mock on line 300.

Reply:
I was wondering about that, is there any better way at all to test this error? Or should this test just be cut entirely from the automated testing like we initially talked about?

Reply:
I think [patch object](https://docs.python.org/3/library/unittest.mock.html#patch-object) does something close to what we need.